vignettes/dm-introduction-relational-data-models.Rmd

```diff
@@ -58,43 +96,86 @@ The links between the tables are established through [primary keys](#pk) and [fo
+You can get all primary keys in a `dm` by calling `cdm_get_all_pks()`:
+
+```{r}
+dm %>%
+  cdm_get_all_pks()
+```
 
-## 3. Foreign Keys {#fk}
+If an attribute is suitable as a primary key, it can be checked with `cdm_enum_pk_candidates()`.
+Which columns of the `airlines` table can serve as a primary key?
+
+```{r}
+dm %>% 
+  cdm_enum_pk_candidates(airports)
+```
+
+Further Reading: The {dm} package offers several function for dealing with [primary keys](https://krlmlr.github.io/dm/articles/dm-class-and-basic-operations.html#pk).
+
+## 4. Foreign Keys {#fk}
 
 The **counterpart of a primary key in one table is the foreign key in another table**.
 In order to join two tables, the primary key of the first table needs to be available in the second table, too.
 This second column is called the foreign key (abbreviated with fk).
 
 For example: If you want to link the `airlines` table to the `flights` table, the primary key in `airlines` needs to match the foreign key in `flights`.
-This is true, because the column `carrier` is present as a primary key in the `airlines` table as well as a foreign key in the `flights` table.
+This is true, because the column `carrier` is present as a primary key in the `airlines` table as well as a foreign key in the `flights` table:
+
+```{r}
+dm %>% 
+  cdm_get_fk(flights, airlines)
+```
 
-Further Reading: The {dm} functions for working with [foreign keys](https://krlmlr.github.io/dm/articles/dm-class-and-basic-operations.html#foreign-keys).
 
+Further Reading: All {dm} functions for working with [foreign keys](https://krlmlr.github.io/dm/articles/dm-class-and-basic-operations.html#foreign-keys).
 
-## 4. Normalisation {#normalisation}
+## 5. Normalisation {#normalisation}
 
 Normalisation is the technical term that describes the **central design principle of a relational data model**:
 splitting data into multiple tables.
 
 A normalized data schema consists of several relations (tables) that are linked with attributes (columns).
-The joining is with [primary](#pk) and [foreign keys](#fk).
+The joining works with [primary](#pk) and [foreign keys](#fk).
 One main goal is to keep the data organization as clean and simple as possible by avoiding redundant data entries.
 
 For example: If you want to change the name of one airport in the `nycflights13` dataset, you have to update just a single data entry.
 Sometimes, this principle is called "single point of truth".
 
+```{r}
+#  Update in one single location...
+airlines[airlines$carrier == "UA", "name"] <- "United broke my guitar"
+
+airlines %>%
+  filter(carrier == "UA")
+
+# ...propagates to all related records
+flights %>%
+  left_join(airlines) %>% 
+  select(flight, name)
+```
+
+Another way to demonstrate normalisation is the function `decompose_table()`.
+It splits one table into two.
+Let's take the `planes` table and determine a new table containing the specified columns.
```

Comment:
Why do we do this? Discuss row count in result.

Reply:
see new commit